### ISSUE TITLE
CSmartTable table & column filters won't work if data loaded after setup

### ISSUE THREAD

**Issue by dang-cb** [coreui/coreui-vue#181]
(Version v4.1.1)

If you define a CSmartTable and load the `items` data via an API call (i.e. data is not already present at the time the CSmartTable component is instanced) then the table and column filters will not work.

These filters leverage `itemsDataColumns` which is defined at setup, and which compares against the keys found in `items.value[0]`.

### Example:
```
<CSmartTable cleaner column-filter table-filter :items="items" :columns="columns" :loading="is_loading">

<script>
export default {
  data() {
    return {
      is_loading: true,
      items: [],
      columns: [
        { key: 'citation' },
        { key: 'title' },
      ],
    };
  },
  mounted() {
    api.load().then((r) => {
      this.is_loading = false;
      this.items = r.data;
    });
  },
}
</script>
```

The table will load the data, but any time a character is entered into a filter the table will blank out with 0 matches because no field keys can be found.

### Workaround:
Give `items` an initial dummy record with the shape of your data, e.g.:
```
items: [{ id: 0, citation: '', title: '' }],
```

**Comment by chuckk589** [comment]
> (Version v4.1.1)
> 
> If you define a CSmartTable and load the `items` data via an API call (i.e. data is not already present at the time the CSmartTable component is instanced) then the table and column filters will not work.
> 
> These filters leverage `itemsDataColumns` which is defined at setup, and which compares against the keys found in `items.value[0]`.
> 
> ### Example:
> ```
> <CSmartTable cleaner column-filter table-filter :items="items" :columns="columns" :loading="is_loading">
> 
> <script>
> export default {
>   data() {
>     return {
>       is_loading: true,
>       items: [],
>       columns: [
>         { key: 'citation' },
>         { key: 'title' },
>       ],
>     };
>   },
>   mounted() {
>     api.load().then((r) => {
>       this.is_loading = false;
>       this.items = r.data;
>     });
>   },
> }
> </script>
> ```
> 
> The table will load the data, but any time a character is entered into a filter the table will blank out with 0 matches because no field keys can be found.
> 
> ### Workaround:
> Give `items` an initial dummy record with the shape of your data, e.g.:
> 
> ```
> items: [{ id: 0, citation: '', title: '' }],
> ```

I dont know how you made it work to this point , i cloned coreui-pro-vue-admin-template-v4.1.0 example and this code is not working

```
<template>
    <CSmartTable :items="items" :columns="columns"/>
</template>

<script>
export default {
  data () {
    return {
      columns:[ {key:"test"} ],
      items: []
    }
  },
  mounted:function() {
    this.items = [ {test:1}, {test:2} ]
  },
}
</script>
```

Table doesnt render any changes that were made to "items" after setup, wasted hours , am i missing something?

**Comment by mrholek** [comment]
Fixed in v4.1.3

**Comment by dang-cb** [comment]
Confirmed working in v4.1.3, thanks @mrholek !
